### ISSUE TITLE
(t) legacy binary path in some scripts

### ISSUE THREAD

**Issue by FroggyFlox** [rockstor/rockstor-core#2470]
On an RPM-installed Rockstor 4.5.4-0, we see the following error when trying to run the `debug-mode` script:
```
$ poetry run debug-mode ON
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/opt/rockstor/src/rockstor/scripts/debugmode.py", line 64, in main
    run_command([SUPERCTL_BIN, "restart", "gunicorn"])
  File "/opt/rockstor/src/rockstor/system/osi.py", line 217, in run_command
    raise Exception("Exception while running command({}): {}".format(cmd, e))
Exception: Exception while running command(['/opt/rockstor/bin/supervisorctl', 'restart', 'gunicorn']): [Errno 2] No such file or directory
```

This is due to the following now-erroneous path defined:
https://github.com/rockstor/rockstor-core/blob/28292f5b0bd1dc61673818d69164d3f471fe15e3/src/rockstor/scripts/debugmode.py#L32

This should be using our poetry venv instead.
[phillxnet: EDIT]  instances found to date summarised:

- [x] https://github.com/rockstor/rockstor-core/blob/ccd21f7703d8974282bc05d232efc8f8c67b7700/src/rockstor/scripts/debugmode.py#L32
- [x] https://github.com/rockstor/rockstor-core/blob/eeb0084a2ea6f6bdb1cdbe35428aa5021ddadc72/src/rockstor/smart_manager/views/docker_service.py#L142
- [x] https://github.com/rockstor/rockstor-core/blob/ccd21f7703d8974282bc05d232efc8f8c67b7700/src/rockstor/smart_manager/views/replication.py#L55
- [x] https://github.com/rockstor/rockstor-core/blob/28292f5b0bd1dc61673818d69164d3f471fe15e3/src/rockstor/smart_manager/views/task_scheduler.py#L104
- [x] https://github.com/rockstor/rockstor-core/blob/28292f5b0bd1dc61673818d69164d3f471fe15e3/src/rockstor/smart_manager/views/task_scheduler.py#L110
- [x] https://github.com/rockstor/rockstor-core/blob/28292f5b0bd1dc61673818d69164d3f471fe15e3/src/rockstor/smart_manager/views/task_scheduler.py#L116
- [x] https://github.com/rockstor/rockstor-core/blob/28292f5b0bd1dc61673818d69164d3f471fe15e3/src/rockstor/storageadmin/views/oauth_app.py#L87
- [x] https://github.com/rockstor/rockstor-core/blob/28292f5b0bd1dc61673818d69164d3f471fe15e3/src/rockstor/storageadmin/templates/storageadmin/login.html#L141

**Comment by FroggyFlox** [comment]
Another instance:
https://github.com/rockstor/rockstor-core/blob/eeb0084a2ea6f6bdb1cdbe35428aa5021ddadc72/src/rockstor/smart_manager/views/docker_service.py#L142

This is legacy code now, though, and should not be affected by the Poetry move.

**Comment by FroggyFlox** [comment]
https://github.com/rockstor/rockstor-core/blob/eeb0084a2ea6f6bdb1cdbe35428aa5021ddadc72/src/rockstor/smart_manager/views/replication.py#L56

This one should be corrected

**Comment by FroggyFlox** [comment]
https://github.com/rockstor/rockstor-core/blob/28292f5b0bd1dc61673818d69164d3f471fe15e3/src/rockstor/smart_manager/views/task_scheduler.py#L104
https://github.com/rockstor/rockstor-core/blob/28292f5b0bd1dc61673818d69164d3f471fe15e3/src/rockstor/smart_manager/views/task_scheduler.py#L110
https://github.com/rockstor/rockstor-core/blob/28292f5b0bd1dc61673818d69164d3f471fe15e3/src/rockstor/smart_manager/views/task_scheduler.py#L116

... should be corrected.

**Comment by FroggyFlox** [comment]
https://github.com/rockstor/rockstor-core/blob/28292f5b0bd1dc61673818d69164d3f471fe15e3/src/rockstor/storageadmin/views/oauth_app.py#L87

... should be corrected.

**Comment by FroggyFlox** [comment]
https://github.com/rockstor/rockstor-core/blob/28292f5b0bd1dc61673818d69164d3f471fe15e3/src/rockstor/storageadmin/templates/storageadmin/login.html#L141

... should be corrected.

**Comment by phillxnet** [comment]
We also have the following comments on running subsets of our tests:

Extracted form "

> src/rockstor/fs/tests/test_btrfs.py:    ./bin/test --settings=test-settings -v 3 -p test_btrfs*
...
> src/rockstor/system/tests/test_osi.py:    ./bin/test --settings=test-settings -v 3 -p test_osi*
> src/rockstor/system/tests/test_pkg_mgmt.py:    ./bin/test --settings=test-settings -v 3 -p test_pkg_mgmt*
> src/rockstor/system/tests/test_pkg_mgmt.py:        #     '/usr/bin/zypper', '--non-interactive', '-q', 'list-updates']
> src/rockstor/system/tests/test_directory_services.py:    ./bin/test --settings=test-settings -v 3 -p test_directory_services*
> src/rockstor/system/tests/test_directory_services.py:            cmd=["/usr/bin/net", "ads", "workgroup", "-S", domain],
> src/rockstor/system/tests/test_system_network.py:    ./bin/test --settings=test-settings -v 3 -p test_system_network*
...

> src/rockstor/storageadmin/tests/test_appliances.py:    # bin/django dumpdata storageadmin.appliance --natural-foreign --indent 4 >
> src/rockstor/storageadmin/tests/test_appliances.py:    # ./bin/test -v 2 -p test_appliances.py
> src/rockstor/storageadmin/tests/test_disk_smart.py:    # bin/django dumpdata storageadmin.disk storageadmin.smartcapability
> src/rockstor/storageadmin/tests/test_disk_smart.py:    # ./bin/test -v 2 -p test_disk_smart.py
> src/rockstor/storageadmin/tests/test_disks.py:            "Error running a command. cmd = /sbin/btrfs fi show "
> src/rockstor/storageadmin/tests/test_pool_balance.py:bin/django dumpdata storageadmin.pool storageadmin.disk storageadmin.poolbalance \
> src/rockstor/storageadmin/tests/test_pool_balance.py:./bin/test -v 2 -p test_pool_balance.py
> src/rockstor/storageadmin/tests/test_pool_balance_huey.py:bin/django dumpdata storageadmin.pool storageadmin.disk storageadmin.poolbalance \
> src/rockstor/storageadmin/tests/test_pool_balance_huey.py:./bin/test -v 2 -p test_pool_balance.py
> src/rockstor/storageadmin/tests/test_pool_scrub.py:bin/django dumpdata storageadmin.pool --natural-foreign --indent 4 > \
> src/rockstor/storageadmin/tests/test_pool_scrub.py:./bin/test -v 2 -p test_pool_scrub.py
> src/rockstor/storageadmin/tests/test_pools.py:bin/django dumpdata storageadmin.pool storageadmin.disk storageadmin.share \
> src/rockstor/storageadmin/tests/test_pools.py:./bin/test -v 2 -p test_pools.py
> src/rockstor/storageadmin/tests/test_samba.py:    # bin/django dumpdata storageadmin.pool storageadmin.share
> src/rockstor/storageadmin/tests/test_samba.py:    # ./bin/test -v 2 -p test_samba.py
> src/rockstor/storageadmin/tests/test_shares.py:bin/django dumpdata storageadmin.pool storageadmin.share storageadmin.snapshot \
> src/rockstor/storageadmin/tests/test_shares.py:bin/django dumpdata --database smart_manager smart_manager.service \
> src/rockstor/storageadmin/tests/test_shares.py:./bin/test -v 2 -p test_shares.py
> src/rockstor/storageadmin/tests/test_snapshot.py:bin/django dumpdata storageadmin.pool storageadmin.share storageadmin.snapshot \
> src/rockstor/storageadmin/tests/test_snapshot.py:./bin/test -v 2 -p test_snapshot.py
> src/rockstor/storageadmin/tests/test_api.py:    # bin/django dumpdata --natural-foreign --indent 4 auth.user > src/rockstor/storageadmin/fixtures/test_api.json
> src/rockstor/storageadmin/tests/test_user.py:        #         'shell': '/bin/xyz', 'email': 'admin2@xyz.com'}
> src/rockstor/storageadmin/tests/test_network.py:    # ./bin/test -v 2 -p test_network.py
..

> src/rockstor/smart_manager/tests/test_task_scheduler.py:    # bin/django dumpdata auth.user storageadmin.pool storageadmin.share
> src/rockstor/smart_manager/tests/test_task_scheduler.py:    # ./bin/test -v 2 -p test_task_scheduler.py
..


We have also a shell definition here:

> src/rockstor/settings.py:VALID_SHELLS = ('{}/bin/rcli'.format(BASE_DIR), '/bin/bash', '/sbin/nologin',)

This file is thought to be redundant but should be removed under it's own issue to prove this claim.
> etc/init.d/gunicorn:DAEMON=/opt/rockstor/bin/gunicorn

**Comment by phillxnet** [comment]
A follow up issue has now been created and linked back to this one concerning the many comment changes required within our test. As these also require other modifications given we no longer maintain a test-settings.py

**Comment by phillxnet** [comment]
I have included, by way of establishing an in-code reference for the future, a fix for the following single test subset instructional comment:

- [ ] https://github.com/rockstor/rockstor-core/blob/ccd21f7703d8974282bc05d232efc8f8c67b7700/src/rockstor/fs/tests/test_btrfs.py#L53

**Comment by phillxnet** [comment]
Closing as:
Fixed by #2476 

Thanks again to @FroggyFlox for much of the ground work on this one.
